File: shopxin/lib/models/product.dart
class Product {
  final String id;
  final String name;
  final int price;
  final String description;
  final String category;
  final String image;
  final String sellerId;
  final String sellerName;

  Product({
    required this.id,
    required this.name,
    required this.price,
    required this.description,
    required this.category,
    required this.image,
    required this.sellerId,
    this.sellerName = '',
  });
}

File: backend/models/product.js
const mongoose = require("mongoose");

const productSchema = mongoose.Schema({
  name: { type: String, required: true },
  price: { type: Number, required: true },
  description: { type: String, required: true },
  sellerId: { type: mongoose.Schema.Types.ObjectId,ref: 'User' },
  productImage: { type: String, required: true },
  category: { type: String, required: true }
});

module.exports = mongoose.model("Product", productSchema);
